politeiad/api/v1/v1.go

```diff
@@ -352,3 +354,45 @@ type UserErrorReply struct {
 type ServerErrorReply struct {
 	ErrorCode int64 `json:"code"` // Server error code
 }
+
+// PluginSetting is a structure that holds key/value pairs of a plugin setting.
+type PluginSetting struct {
+	Key   string `json:"key"`   // Name of setting
+	Value string `json:"value"` // Value of setting
+}
+
+// Plugin describes a plugin and its settings.
+type Plugin struct {
+	ID       string          `json:"id"`      // Identifier
+	Version  string          `json:"version"` // Version
+	Settings []PluginSetting `json:"setting"` // Settings
+}
+
+// PluginInventory retrieves all active plugins and their settings.
+type PluginInventory struct {
+	Challenge string `json:"challenge"` // Random challenge
+}
+
+// PluginInventoryReply returns all plugins and their settings.
+type PluginInventoryReply struct {
+	Response string   `json:"response"` // Challenge response
+	Plugins  []Plugin `json:"plugins"`  // Plugins and their settings
+}
+
+// PluginCommand sends a command to a plugin.
+type PluginCommand struct {
+	Challenge string `json:"challenge"` // Random challenge
+	ID        string `json:"id"`        // Plugin identifier
+	Command   string `json:"command"`   // Command identifier
+	CommandID string `json:"commandid"` // User setable command identifier
+	Payload   string `json:"payload"`   // Actual command
+}
+
+// PluginCommandReply is the reply to a PluginCommand.
+type PluginCommandReply struct {
+	Response  string `json:"response"`  // Challenge response
+	ID        string `json:"id"`        // Plugin identifier
+	Command   string `json:"command"`   // Command identifier
+	CommandID string `json:"commandid"` // User setable command identifier
+	Payload   string `json:"payload"`   // Actual command reply
```

Comment:
Does the reply need to include `ID`, `Command` and `CommandID` if the client already has that info?

Reply:
Yes, there can be multiple plugins.